addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/internal/discovery/VeraBridgeDiscoveryService.java

```diff
@@ -0,0 +1,142 @@
+                if (networkInterface.isUp() && !networkInterface.isVirtual() && !networkInterface.isLoopback()) {
+                    for (InterfaceAddress address : networkInterface.getInterfaceAddresses()) {
+                        if (validator.isValidIPV4(address.getAddress().getHostAddress())) {
+                            String ipAddress = address.getAddress().getHostAddress();
+                            Short prefix = address.getNetworkPrefixLength();
+
+                            logger.debug("Scan IP address for Vera Controller: {}", ipAddress);
+
+                            String subnet = ipAddress + "/" + prefix;
+                            SubnetUtils utils = new SubnetUtils(subnet);
+                            String[] addresses = utils.getInfo().getAllAddresses();
+
+                            for (String addressInSubnet : addresses) {
+                                scheduler.execute(new VeraControllerScan(addressInSubnet));
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (SocketException e) {
+            logger.warn("Error occurred while searching Vera controller ({})", e.getMessage());
+        }
+    }
+
+    public boolean pingHost(String host, int port, int timeout) {
+        try (Socket socket = new Socket()) {
+            socket.connect(new InetSocketAddress(host, port), timeout);
+            return true;
+        } catch (IOException e) {
+            return false; // Either timeout or unreachable or failed DNS lookup.
+        }
+    }
+
+    public class VeraControllerScan implements Runnable {
+        private String ipAddress;
+
+        public VeraControllerScan(String ipAddress) {
+            this.ipAddress = ipAddress;
+        }
+
+        @Override
+        public void run() {
+            if (!pingHost(ipAddress, 3480, 100)) {
+                return;
+            }
+
+            try {
+                URL url = new URL("http://" + ipAddress + ":3480/data_request?id=sdata&output_format=json");
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+                if (connection.getResponseCode() == 200) {
+                    ThingUID thingUID = new ThingUID(VeraBindingConstants.THING_TYPE_BRIDGE,
+                            ipAddress.replaceAll("\\.", "_"));
+
+                    // Attention: if is already present as thing in the ThingRegistry
+                    // the configuration for thing will be updated!
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID)
+                            .withProperty(VeraBindingConstants.BRIDGE_CONFIG_VERA_SERVER_IP_ADDRESS, ipAddress)
+                            .withLabel("Vera controller " + ipAddress).build();
+                    thingDiscovered(discoveryResult);
+                }
+            } catch (Exception e) {

```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception